static/js/components/source/source_publish/SourcePublish.jsx

```diff
@@ -0,0 +1,284 @@
+    if (updateThumbnailsData) {
+      processThumbnailsData();
+    }
+    const getPhotometry = () => {
+      if (!optionsState.include_photometry) {
+        return null;
+      }
+      // filter photometry based on selected groups and streams
+      return isElements.photometry
+        ? photometry.filter(
+            (onePhotometry) =>
+              (!optionsState.groups.length ||
+                onePhotometry.groups.some((group) =>
+                  optionsState.groups.includes(group.id),
+                )) &&
+              (!optionsState.streams.length ||
+                onePhotometry.streams.some((stream) =>
+                  optionsState.streams.includes(stream.id),
+                )),
+          )
+        : [];
+    };
+
+    const getClassifications = () => {
+      if (!optionsState.include_classifications) {
+        return null;
+      }
+      // filter classifications based on selected groups
+      return source.classifications?.filter(
+        (onePhotometry) =>
+          !optionsState.groups.length ||
+          onePhotometry.groups.some((group) =>
+            optionsState.groups.includes(group.id),
+          ),
+      );
+    };
+
+    return {
+      source_id: source.id,
+      radec_hhmmss: source.radec_hhmmss,
+      ra: source.ra,
+      dec: source.dec,
+      gal_lon: source.gal_lon?.toFixed(6),
+      gal_lat: source.gal_lat?.toFixed(6),
+      ebv: source.ebv?.toFixed(2),
+      redshift: source.redshift?.toFixed(source.z_round),
+      dm: source.dm?.toFixed(3),
+      dl: source.luminosity_distance?.toFixed(2),
+      thumbnails: source.thumbnails,
+      photometry: getPhotometry(),
+      classifications: getClassifications(),
+    };
+  };
+
+  const publish = () => {
+    if (!permissionToPublish) return;
+    setPublishButton("loading");
+    const payload = {
+      public_data: publicData(),
+    };
+    dispatch(
+      publicSourcePageActions.generatePublicSourcePage(source.id, payload),
+    ).then((data) => {
+      setPublishButton("done");
+      setNewPageGenerate(data.data.page);
+      setTimeout(() => {
+        setPublishButton("publish");
+      }, 2000);
+    });
+  };
+
+  return (
+    <div>
+      <Button
+        secondary
+        size="small"
+        onClick={() => setSourcePublishDialogOpen(true)}
+      >
+        <Tooltip title="Click here if you want to see the public access information">
+          <span>Public access</span>
+        </Tooltip>
+      </Button>
+      <Dialog
+        open={sourcePublishDialogOpen}
+        onClose={() => setSourcePublishDialogOpen(false)}
+        PaperProps={{ style: { maxWidth: "700px" } }}
+      >
+        <DialogTitle>Public access information</DialogTitle>
+        <DialogContent style={{ paddingBottom: "0.5rem" }}>
+          <div style={{ marginBottom: "1rem" }}>
+            You are about to change the public access settings for this source
+            page. This information will be available to everyone on the
```

Comment:
maybe  "The data you selected" rather than info?

Reply:
✅ Done.